pandas/core/arrays/boolean.py

```diff
@@ -0,0 +1,679 @@
+
+        name = "__{name}__".format(name=op.__name__)
+        return set_function_name(logical_method, name, cls)
+
+    @classmethod
+    def _create_comparison_method(cls, op):
+        op_name = op.__name__
+
+        def cmp_method(self, other):
+
+            if isinstance(other, (ABCDataFrame, ABCSeries, ABCIndexClass)):
+                # Rely on pandas to unbox and dispatch to us.
+                return NotImplemented
+
+            other = lib.item_from_zerodim(other)
+            mask = None
+
+            if isinstance(other, BooleanArray):
+                other, mask = other._data, other._mask
+
+            elif is_list_like(other):
+                other = np.asarray(other)
+                if other.ndim > 1:
+                    raise NotImplementedError(
+                        "can only perform ops with 1-d structures"
+                    )
+                if len(self) != len(other):
+                    raise ValueError("Lengths must match to compare")
+
+            # numpy will show a DeprecationWarning on invalid elementwise
+            # comparisons, this will raise in the future
+            with warnings.catch_warnings():
+                warnings.filterwarnings("ignore", "elementwise", FutureWarning)
+                with np.errstate(all="ignore"):
+                    result = op(self._data, other)
+
+            # nans propagate
+            if mask is None:
+                mask = self._mask
+            else:
+                mask = self._mask | mask
+
+            result[mask] = op_name == "ne"
+            return BooleanArray(result, np.zeros(len(result), dtype=bool), copy=False)
+
+        name = "__{name}__".format(name=op.__name__)
+        return set_function_name(cmp_method, name, cls)
+
+    def _reduce(self, name, skipna=True, **kwargs):
+        data = self._data
+        mask = self._mask
+
+        # coerce to a nan-aware float if needed
+        if mask.any():
+            data = self._data.astype("float64")
+            data[mask] = self._na_value
+
+        op = getattr(nanops, "nan" + name)
+        result = op(data, axis=0, skipna=skipna, mask=mask)
+
+        # if we have a boolean op, don't coerce
+        if name in ["any", "all"]:
+            pass
+
+        # if we have a preservable numeric op,
+        # provide coercion back to an integer type if possible
+        elif name in ["sum", "min", "max", "prod"] and notna(result):
+            int_result = int(result)
+            if int_result == result:
+                result = int_result
+
+        return result
+
+    def _maybe_mask_result(self, result, mask, other, op_name):
+        """
+        Parameters
+        ----------
+        result : array-like
+        mask : array-like bool
+        other : scalar or array-like
+        op_name : str
+        """
+
+        # may need to fill infs
+        # and mask wraparound
+        if is_float_dtype(result):
+            mask |= (result == np.inf) | (result == -np.inf)
```

Comment:
This looks like it is copied from IntegerArray, and should not be an inplace operation.  It risks alterting self._mask inplace. xref #27829

Reply:
> This looks like it is copied from IntegerArray, and should not be an inplace operation. It risks alterting self._mask inplace. xref #27829

OK, I removed this here altogether, as I don't think we should fill any infs in the BooleanArray division (we should separately also look at this for IntegerArray to see what behaviour we want)

This means that for BooleanArray I follow numpy's behaviour in returning float with inf:

```
In [19]: True / np.array([True, False, True])
/home/joris/miniconda3/envs/dev/bin/ipython:1: RuntimeWarning: divide by zero encountered in true_divide
  #!/home/joris/miniconda3/envs/dev/bin/python
Out[19]: array([ 1., inf,  1.])
```